pkg/storage/driver/util.go

```diff
@@ -24,24 +24,29 @@ import (
 	"io"
 	"slices"
 
-	rspb "helm.sh/helm/v4/pkg/release/v1"
+	"github.com/klauspost/compress/zstd"
+
+	rspb "helm.sh/helm/v3/pkg/release"
 )
 
 var b64 = base64.StdEncoding
 
-var magicGzip = []byte{0x1f, 0x8b, 0x08}
+var (
+	magicGzip = []byte{0x1f, 0x8b, 0x08}
+	magicZstd = []byte{0x28, 0xb5, 0x2f, 0xfd}
+)
 
 var systemLabels = []string{"name", "owner", "status", "version", "createdAt", "modifiedAt"}
 
 // encodeRelease encodes a release returning a base64 encoded
-// gzipped string representation, or error.
+// zstded string representation, or error.
 func encodeRelease(rls *rspb.Release) (string, error) {
 	b, err := json.Marshal(rls)
 	if err != nil {
 		return "", err
 	}
 	var buf bytes.Buffer
-	w, err := gzip.NewWriterLevel(&buf, gzip.BestCompression)
+	w, err := zstd.NewWriter(&buf, zstd.WithEncoderLevel(zstd.SpeedBestCompression))
```

Comment:
Lets leave the default as gzip 

https://github.com/helm/helm/pull/31232#issuecomment-3331408963